### PR TITLE
HDDS-8300. Enable topology aware replication and reads by default.

### PR DESCRIPTION
**Pull request by ashishkumar50**
## What changes were proposed in this pull request?

Enable topology aware replication and read by default.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8300

## How was this patch tested?

Test using TestSCMContainerPlacementRackAware integration test.

### CODE REVIEW

File: hadoop-hdds/common/src/main/resources/ozone-default.xml [hunk 1]
```diff
@@ -2753,7 +2753,7 @@
   </property>
   <property>
     <name>ozone.network.topology.aware.read</name>
-    <value>false</value>
+    <value>true</value>
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/resources/ozone-default.xml:2756]: Default value should also be updated at:

https://github.com/apache/ozone/blob/cce3c7ea79ec19657d10f011ddeaf3067a2f83bd/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java#L487

**Comment by adoroszlai** [comment]
Topology-aware read breaks EC file checksums (HDDS-8276).  I have converted this to draft, because we need to fix that first.

**Comment by adoroszlai** [comment]
@ashishkumar50 After fixing HDDS-8276, the following problems are left:

 * `TestXceiverClientGrpc` is [timing out](https://github.com/ashishkumar50/ozone/actions/runs/4568058758/jobs/8062638544#step:5:3660)
 * `ozone debug read-replicas` test is [failing in _acceptance (unsecure)_](https://github.com/ashishkumar50/ozone/actions/runs/4568058758/jobs/8063070832#step:5:1015)

Can you please check?

**Comment by ashishkumar50** [comment]
@adoroszlai , TestXceiverClientGrpc test issue is fixed, we need to disable topology read for shuffle read from DN which is used in testRandomFirstNodeIsCommandTarget case.
Another test case testFirstNodeIsCorrectWithTopologyForCommandTarget which is meant for sortedDN always.

**Comment by adoroszlai** [comment]
> * `ozone debug read-replicas` test is [failing in _acceptance (unsecure)_](https://github.com/ashishkumar50/ozone/actions/runs/4568058758/jobs/8063070832#step:5:1015)

Filed HDDS-8364 for the bug in `read-replicas`, being fixed in #4522.

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/TestContainerPlacementFactory.java [hunk 2]
```diff
@@ -163,10 +163,13 @@ public void testRackAwarePolicy() throws IOException {
...
-  public void testDefaultPolicy() throws IOException {
+  public void testRackAwareContainerPolicy() throws IOException {
+    conf.set(ScmConfigKeys.OZONE_SCM_CONTAINER_PLACEMENT_IMPL_KEY,
+            SCMContainerPlacementRackAware.class.getName());
     PlacementPolicy policy = ContainerPlacementPolicyFactory
         .getPolicy(conf, null, null, true, null);
-    Assertions.assertSame(SCMContainerPlacementRandom.class, policy.getClass());
+    Assertions.assertSame(SCMContainerPlacementRackAware.class,
+            policy.getClass());
   }
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/TestContainerPlacementFactory.java:173]: I would have preferred keeping the intention of this test: testing default setting.  Sorry for not noticing this earlier.

**Review by adoroszlai — APPROVED** [review]

**Review by adoroszlai — DISMISSED** [review]
Sorry @ashishkumar50 for the late notice, but after checking results of `TestDecommissionAndMaintenance`, it seems `testContainerIsReplicatedWhenAllNodesGotoMaintenance` is consistently failing (timing out).

**Comment by adoroszlai** [comment]
Created HDDS-8383 for the bug revealed by `TestDecommissionAndMaintenance#testContainerIsReplicatedWhenAllNodesGotoMaintenance`, being fixed in #4539.

**Review by adoroszlai — APPROVED** [review]

**Pull request merged**